Tensile/KernelWriterAssembly.py

```diff
@@ -579,36 +589,35 @@ def getLinkCodeObjectArgs(self, objectFileNames, coFileName, *moreArgs):
     return rv
 
   ########################################
-  def getOccupancy(self, kernel, vgprs, accvgprs=0):
+  @staticmethod
+  def getOccupancy(kernel, vgprs, accvgprs=0):
     multiplier = int(ceil(max(kernel["NumThreads"], 256) / 256.0))
     # example: wg=512 multiplier=2, 1024=4
 
     ldsSize = kernel["LdsNumElements"] * kernel["ProblemType"]["DataType"].numBytes()
     ldsSize = (ldsSize + 255) & 0x1ff00 # 256-byte granularity
-    ldsLimitedOccupancy = self.getLdsLimitedOccupancy(ldsSize)
+    ldsLimitedOccupancy = KernelWriterAssembly.getLdsLimitedOccupancy(ldsSize)
 
     vgprs *= multiplier
-    vgprLimitedOccupancy =  self.vgprOccupancy[vgprs] if vgprs <= 256 else 0
+    vgprLimitedOccupancy =  vgprOccupancy[vgprs]
 
     accvgprs *= multiplier
-    accvgprLimitedOccupancy =  self.vgprOccupancy[accvgprs] if accvgprs <= 256 else 0
+    accvgprLimitedOccupancy =  vgprOccupancy[accvgprs]
 
     return min(ldsLimitedOccupancy, vgprLimitedOccupancy, accvgprLimitedOccupancy)
 
   # TODO: also consider sgpr
-  def getMaxRegsForOccupancy(self, vgprs, ldsSize, accvgprs=0):
-
-    vgprOccupancy = self.vgprOccupancy[vgprs] if vgprs <= 256 else 0
-    accvgprOccupancy = self.vgprOccupancy[accvgprs] if accvgprs <= 256 else 0
-    initOccupancy = vgprOccupancy
+  @staticmethod
+  def getMaxRegsForOccupancy(vgprs, ldsSize, accvgprs=0):
+    initOccupancy = vgprOccupancy[vgprs]
     if accvgprs > 0:
-      initOccupancy = min(initOccupancy, accvgprOccupancy)
+      initOccupancy = min(initOccupancy, vgprOccupancy[accvgprs])
     if ldsSize > 0:
-      initOccupancy = min(initOccupancy, self.getLdsLimitedOccupancy(ldsSize))
+      initOccupancy = min(initOccupancy, KernelWriterAssembly.getLdsLimitedOccupancy(ldsSize))
     lastVgprs = vgprs
-    while vgprs < len(self.vgprOccupancy)-1:
+    while vgprs < len(vgprOccupancy)-1:
```

Comment:
using if condition or a static array should be able to get the largest vgpr within the same occupancy criteria rather than loop search.

Reply:
@imcarsonliao I would prefer the vgpr#-occupancy curve to be the single source of truth. I simply want to avoid defining another lookup table for this specific use case.